root/static/scripts/relationship-editor/components/RelationshipDialogContent.js

```diff
@@ -322,6 +377,13 @@ export function reducer(
           action.source,
         );
       }
+
+      const oldTargetGid = state.targetEntity.target.gid;
+      const newTargetGid = newState.targetEntity.target.gid;
+      if (oldTargetGid !== newTargetGid) {
```

Comment:
For reasons that I don't understand (unrelated to this change), `reducer` appears to be called multiple times whenever a change is made. For example, when I select an entity, I see two seemingly-identical `update-target-entity` calls come in changing the target from null to the selected entity. Is that expected?

Reply:
I didn’t find why so it’s probably preferable to document it in the code itself as follows.

```suggestion
      // For some reason this update function may be called twice for a single change
      if (oldTargetGid !== newTargetGid) {
```

Reply:
Just to clarify, here are all the updates that I'm seeing:

When first clicking "Add relationship":
- update-link-type (no change in type)
- update-target-entity (null to null)
- update-link-type (no change)
- update-target-entity (null to null)

When clicking the "Relationship type" selector to open the dropdown:
- update-link-type (no change)
- update-link-type (no change)

When hovering over the "member of" item:
- update-link-type (no change)
- update-link-type (no change)

When clicking "member of":
- update-link-type (type changed)
- update-link-type (type changed)

When clicking the "Artist" selector to open the dropdown:
- update-target-entity (null to null)
- update-target-entity (null to null)

When hovering over an artist:
- update-target-entity (null to null)
- update-target-entity (null to null)

When clicking on an artist:
- update-target-entity (null to selected MBID)
- update-target-entity (null to selected MBID)

So this check prevents unnecessary calls in the null-to-null update-target-entity updates, but even when the type or artist actually changes I still hear about it twice, and I'm not sure if there's any way to detect that case. I suspect that something may be set up incorrectly on the React side, but I don't have enough experience with React to debug that any further.

Reply:
I've added a comment describing what I'm seeing.

Reply:
At a glance it sounds like the [expected behavior of strict mode in development](https://react.dev/reference/react/StrictMode#reference).

Reply:
Thanks, that was it -- I no longer see the duplicate actions after temporarily commenting out the strict mode tags. I've updated this comment to just refer to the null-to-null updates that the if statement is skipping.